Common/cpp/SharedItems/ShareableValue.cpp

```diff
@@ -94,19 +94,25 @@ void ShareableValue::adapt(jsi::Runtime &rt, const jsi::Value &value, ValueType
         type = ValueType::HostFunctionType;
         hostRuntime = &rt;
         hostFunction = std::make_shared<HostFunctionHandler>(std::make_shared<jsi::Function>(object.asFunction(rt)), rt);
+        containsHostFunction = true;
       } else {
         // a worklet
         type = ValueType::WorkletFunctionType;
         frozenObject = std::make_shared<FrozenObject>(rt, object, module);
+        containsHostFunction |= frozenObject->containsHostFunction;
         if (isRNRuntime) {
-          addHiddenProperty(rt, createHost(rt, frozenObject), object, HIDDEN_HOST_OBJECT_PROP);
+          if (!containsHostFunction) {
```

Comment:
Can we change this to `if (isRNRuntime && !containsHostFunction)` ? - micro optimization because there can exist only one condition jump, however compiler should change this to one jump also.

Reply:
fixed 93ce23b